src/sessions/video-session-handler.ts

```diff
@@ -290,19 +294,39 @@ export default class VideoSessionHandler extends BaseSessionHandler {
     this.setupTransceivers(session);
 
     const attachParams = { audioElement, videoElement };
-    if (session.tracks.length) {
-      session.tracks.forEach((track) => {
-        this.log(LogLevels.info, 'Incoming track', { track, conversationId: session.conversationId });
+
+    const handleIncomingTracks = (session: IJingleSession, tracks: MediaStreamTrack | MediaStreamTrack[]) => {
+      if (!Array.isArray(tracks)) tracks = [tracks];
+
+      for (const track of tracks) {
+        this.log(LogLevels.info, 'Incoming track', {
+          track,
+          conversationId: session.conversationId,
+          sessionId: session.id
+        });
+
         const el = this.attachIncomingTrackToElement(track, attachParams);
-        if (el instanceof HTMLAudioElement) session._outputAudioElement = el;
-      });
+
+        /* if the track was attatched to an audio element, we have an audio track */
+        if (el instanceof HTMLAudioElement) {
+          session._outputAudioElement = el;
+
+          /* if we have support, make sure to set the sinkId on the element */
+          if (hasOutputDeviceSupport()) {
```

Comment:
I think it might make sense to move this block to the base-session-handler so it affects softphone also.